components/net/cookie.rs

```diff
@@ -139,6 +149,9 @@ impl Cookie {
 
     // http://tools.ietf.org/html/rfc6265#section-5.1.3
     pub fn domain_match(string: &str, domain_string: &str) -> bool {
+        let string = &string.to_lowercase();
+        let domain_string = &domain_string.to_lowercase();
```

Comment:
tests/net/cookie_http_state.rs::test_domain0026 uses the domain eXaMpLe.org, which causes a failure in domain match--putting to_lowercase() in the relevant locations solved it